tasks/modules.py

```diff
@@ -403,3 +404,43 @@ def validate(_: Context):
         message += "Please add them to the DEFAULT_MODULES list or exclude them from the validation."
 
         raise Exit(message)
+
+
+@task
+def validate_used_by_otel(ctx: Context):
+    """
+    Verify whether indirect local dependencies of modules labeled "used_by_otel" are also marked with the "used_by_otel" tag.
+    """
+    otel_mods = [path for path, module in DEFAULT_MODULES.items() if module.used_by_otel]
+    missing_used_by_otel_label: dict[str, list[str]] = {}
+
+    # for every module labeled as "used_by_otel"
+    for otel_mod in otel_mods:
+        gomod_path = f"{otel_mod}/go.mod"
+        # get the go.mod data
+        result = ctx.run(f"go mod edit -json {gomod_path}", hide='both')
+        if result.failed:
+            raise Exception(f"Error running go mod edit -json on {gomod_path}: {result.stderr}")
+
+        go_mod_json = json.loads(result.stdout)
+        # get module dependencies
+        reqs = go_mod_json.get("Require", [])
+        if not reqs:  # Module don't have dependencies, continue
+            continue
+        for require in reqs:
+            # we are only interested into local modules
+            if not require["Path"].startswith("github.com/DataDog/datadog-agent/"):
+                continue
+            rel_path = require['Path'][33:]
+            # check if indirect module is labeled as "used_by_otel"
+            if rel_path not in DEFAULT_MODULES or not DEFAULT_MODULES[rel_path].used_by_otel:
+                if rel_path not in missing_used_by_otel_label:
+                    missing_used_by_otel_label[rel_path] = []
```

Comment:
You can use a [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict) instead of a simple dict and get rid of this `if`

Reply:
Done in ca15a0a86fcf04c4999031c44433b304b6558801.